dashboard/app/models/course.rb

```diff
@@ -24,6 +24,81 @@ def skip_name_format_validation
 
   include SerializedToFileValidation
 
+  def to_param
+    name
+  end
+
+  def self.file_path(name)
+    Rails.root.join("config/courses/#{name}.course")
+  end
+
+  def self.load_from_path(path)
+    serialization = File.read(path)
+    hash = JSON.parse(serialization)
+    course = Course.find_or_create_by!(name: hash['name'])
+    course.update_scripts(hash['script_names'])
+  rescue Exception => e
+    # print filename for better debugging
+    new_e = Exception.new("in course: #{path}: #{e.message}")
+    new_e.set_backtrace(e.backtrace)
+    raise new_e
+  end
+
+  # Updates courses.en.yml with our new localizeable strings
+  # @param name [string] - name of the course being updated
+  # @param course_strings[Hash{String => String}]
+  def self.update_strings(name, course_strings)
+    courses_yml = File.expand_path('config/locales/courses.en.yml')
+    i18n = File.exist?(courses_yml) ? YAML.load_file(courses_yml) : {}
+
+    i18n.deep_merge!({'en' => {'data' => {'course' => {'name' => {name => course_strings.to_h}}}}})
+    File.write(courses_yml, "# Autogenerated scripts locale file.\n" + i18n.to_yaml(line_width: -1))
```

Comment:
I've attempted to clean up some of the existing usages (for scripts.en.yml) in the past, and found it quite challenging. I think there is probably opportunity to make this cleaner, but that it's more work than I'm willing to put in right now.

Non-thread safe is a good point, though again this is already a problem with our existing usages for scripts.en.yml (the good news is we expect courses to be edit even more infrequently than scripts).